bootstrap.sh

```diff
@@ -1,10 +1,22 @@
 #!/bin/sh
 
+set -e
+
+export GOBIN=$(pwd)/$(go env GOHOSTOS)_$(go env GOHOSTARCH)/bin
+echo GOBIN is now $GOBIN
+
 echo Building the build tool...
+GO111MODULE=on go get ./util/src/harvey/cmd/...
 
-GOBIN="$(pwd)/util" go get -u harvey-os.org/cmd/ufs
-GO111MODULE=on GOBIN="$(pwd)/util" go get ./util/src/harvey/cmd/...
+echo Fetching u-root and building it...
+GO111MODULE=off go get github.com/u-root/u-root
```

Comment:
why are we turning off modules? `GO111MODULE` doesn't require the target package to have a `.mod` file. 
If it's all the same to you I think we should use modules and pin specific versions of external packages.

Reply:
Also note that Harvey's `go.mod` can influence what version of external dependency is used. Which is why I suggest running `go get` inside a temporary directory that's not part of a module. See https://github.com/golang/go/issues/40276

Reply:
But in this case that is precisely what we want, we always want a known and tested version of external packages to be downloaded and installed. I feel this is important since we don't have testing of u-root packages on a plan9 system. It's very likely that someone might break a u-root package and not even realize it's broken for plan9.

Reply:
Sorry, I meant dependency of u-root can be influence by harvey's dependency when doing the minimum version selection. For example, u-root uses an older version `golang.org/x/tools` package when installed from the harvey module:
```
$ export GO111MODULE=on
$ cd $HARVEY
$ GOBIN=$(pwd)/util/ go get github.com/u-root/u-root@latest
...
$ go version -m util/u-root 
util/u-root: go1.15
	path	github.com/u-root/u-root
	mod	github.com/u-root/u-root	v6.0.0+incompatible	h1:YqPGmRoRyYmeg17KIWFRSyVq6LX5T6GSzawyA6wG6EE=
	dep	github.com/dustin/go-humanize	v1.0.0	h1:VSnTsYCnlFHaM2/igO1h6X3HA71jcobQuxemgkq4zYo=
	dep	golang.org/x/sys	v0.0.0-20190412213103-97732733099d	h1:+R4KGOnez64A81RvjARKc4UT5/tI9ujCIVX+P5KiHuI=
	dep	golang.org/x/tools	v0.0.0-20190524140312-2c0ae7006135	h1:5Beo0mZN8dRzgrMMkDp0jc8YXQKx9DiJ2k1dkvGsn5A=


$ mkdir /tmp/foo
$ cd /tmp/foo/
$ GOBIN=$(pwd) go get github.com/u-root/u-root@latest
...
$ go version -m u-root 
u-root: go1.15
	path	github.com/u-root/u-root
	mod	github.com/u-root/u-root	v6.0.0+incompatible	h1:YqPGmRoRyYmeg17KIWFRSyVq6LX5T6GSzawyA6wG6EE=
	dep	github.com/dustin/go-humanize	v1.0.0	h1:VSnTsYCnlFHaM2/igO1h6X3HA71jcobQuxemgkq4zYo=
	dep	golang.org/x/mod	v0.3.0	h1:RM4zey1++hCTbCVQfnWeKs9/IEsaBLA8vTkd0WVtmH4=
	dep	golang.org/x/sys	v0.0.0-20200819171115-d785dc25833f	h1:KJuwZVtZBVzDmEDtB2zro9CXkD9O0dpCv4o2LHbQIAw=
	dep	golang.org/x/tools	v0.0.0-20200820010801-b793a1359eac	h1:DugppSxw0LSF8lcjaODPJZoDzq0ElTGskTst3ZaBkHI=
	dep	golang.org/x/xerrors	v0.0.0-20191204190536-9bdfabe68543	h1:E7g+9GITq07hpfrRu66IVDexMakfv52eLZ2CXBWiKr4=
```

Reply:
I don't think that's a problem for us. 
One of our goals was (is #319?) to make harvey builds reproducible. 
I feel like using `@latest` in the bootstrap process would cause us to back slide in trying to achieve that goal.  As you've demonstrated it's easy to verify what ever it is we're checking in is correct, if you don't think it's correct, update the mod file so it does the correct thing. It requires a bit more diligence on our part but I'd rather have that than `go get @latest` and have builds randomly break because some external dependency updated.

Reply:
u-root and modules still don't play well. If you've tested this with modules turned on and it works, I'm interested to hear it.

Reply:
did you try it first? I'd like to know that it works. Would be appreciated if you could try a thing before advocating changing that thing.

Reply:
yeah, I did.

case in point with with the modules turned off I get
```
GOBIN is now /home/sevki/src/test/harvey/linux_amd64/bin
Building the build tool...
Fetching u-root and building it...
Fetch harvey-os.org commands and build them into /home/sevki/src/test/harvey/linux_amd64/bin
# harvey-os.org/internal/ufs
../../../go/src/harvey-os.org/internal/ufs/filesystem.go:443:45: undefined: protocol.ListenerOpt
../../../go/src/harvey-os.org/internal/ufs/filesystem.go:443:69: undefined: protocol.Listener
../../../go/src/harvey-os.org/internal/ufs/filesystem.go:458:12: undefined: protocol.NewListener
```
because my `$GOPATH` had `511a25aa7f1542a74d755af15e02503a1d87c71a` checked out which was broken and turning on go modules for all fixed the bootstrap script.

Reply:
thanks, good to know it worked for you, I just pushed it with modules on